### PR TITLE
text viewer mk3 haha!!

### PR DESCRIPTION
**Pull request by nortikin**
## Addressed problem description
Node about viewer text PR. Old approach for trash.

## Solution description
MK3
vtmk3
and https://github.com/nortikin/sverchok/pull/2048/files#diff-0147edc4e7953b77a018b66924e77a37R48 was error giving. name unknown

## Preflight checklist

- [x] Code changes complete.
- [x] Code documentation complete.
- [x] Documentation for users complete (or not required, if user never sees these changes).
- [x] Manual testing done. 
- [x] Unit-tests implemented.
- [x] Ready for merge.

### CODE REVIEW

**Comment by portnov** [comment]
comments.

**Comment by Durman** [comment]
Maybe it will be interesting to have somewhere name of the label of a input node if the label is filled.

**Comment by nortikin** [comment]
test left. how to make... there was issue i guess

File: nodes/text/viewer_text_mk3.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+                if label:
+                    label = '; node ' + label.upper()
+                name = insert.name.upper()
+                # vertices socket
+                if insert.other.bl_idname == 'VerticesSocket':
+                    itype = '\n\nSocket ' + name + label + '; type VERTICES: \n'
+
+                # edges/faces socket
+                elif insert.other.bl_idname == 'StringsSocket':
+                    itype = '\n\nSocket ' + name + label + '; type EDGES/POLYGONS: \n'
```
Comment (portnov) [inline nodes/text/viewer_text_mk3.py]: Huh. @zeffii this is nearly the same as we were talking in #2058: text viewer nodes assumes that StringsSockets is for edges or polygons, but in fact it is for anything for now. @nortikin maybe put a bit more general phrase here, smth like "EDGES/POLYGONS/OTHER" ?

Also I'm not sure about caps :)

File: nodes/text/viewer_text_mk3.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    bl_label = "Sverchok viewer.mk1"
+    bl_options = {'REGISTER', 'UNDO'}
+
+    nodename = StringProperty(name='nodename')
+    treename = StringProperty(name='treename')
+    lines = IntProperty(name='lines', description='lines count for operate on',default=1000)
+
+    def execute(self, context):
+        node = bpy.data.node_groups[self.treename].nodes[self.nodename]
+        inputs = node.inputs
```
Comment (portnov) [inline nodes/text/viewer_text_mk3.py]: I'd propose to extract most logic of this execute() method into separate method, that could be called without operator. Something like `def get_text(input_socket)` or maybe even `def get_text(output_socket)`.

File: nodes/text/viewer_text_mk3.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+from bpy.props import StringProperty, BoolProperty, IntProperty
+
+from sverchok.node_tree import SverchCustomTreeNode
+from sverchok.data_structure import levelsOflist, changable_sockets, multi_socket, updateNode
+
+
+
+class SverchokViewerMK1(bpy.types.Operator):
+    """Sverchok viewerMK1"""
+    bl_idname = "node.sverchok_viewer_buttonmk1"
```
Comment (portnov) [inline nodes/text/viewer_text_mk3.py]: Isn't this the same bl_idname as one that is used in older text viewer node? That could cause funny problems.

File: nodes/text/viewer_text_mk3.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+            a.color[:] = color
+
+    def do_text(self,outs,node):
+        nTree = bpy.data.node_groups[self.treename]
+        #this part can be than removed from node text viewer:
+
+        if not 'Sverchok_viewer' in bpy.data.texts:
+            bpy.data.texts.new('Sverchok_viewer')
+
+        podpis = '\n'*2 \
```
Comment (portnov) [inline nodes/text/viewer_text_mk3.py]: Huh. At least some of Sverchok developers are not russian, so let's rename this to "footer", if you do not mind :)

File: nodes/text/viewer_text_mk3.py (hunk 3 above)
Comment (nortikin) [reply nodes/text/viewer_text_mk3.py]: no

File: nodes/text/viewer_text_mk3.py (hunk 2 above)
Comment (nortikin) [reply nodes/text/viewer_text_mk3.py]: i'll try

**Comment by nortikin** [comment]
what to do with tests?
```bash
TypeError: bpy_prop_collection[key] = value: invalid, expected a StructRNA type or None, not a list
```

**Comment by nortikin** [comment]
so i merge?

**Comment by portnov** [comment]
@nortikin please see #2071 .

**Comment by nortikin** [comment]
will we merge it?

**Comment by portnov** [comment]
Okay now I'm for merging :)

**Pull request merged**